Source/ResponseSerialization.swift

```diff
@@ -377,7 +381,7 @@ extension DownloadRequest {
                         didComplete = { completionHandler(response) }
 
                     case .doNotRetryWithError(let retryError):
-                        let result = Result<T.SerializedObject, Error>.failure(retryError)
+                        let result: Result<T.SerializedObject, AFError> = .failure(retryError.asAFError(default: .requestRetryFailed(retryError: retryError, originalError: serializerError)))
```

Comment:
This should be the `requestRetryFailed` error already, like the other instances.

Reply:
Replaced with an explicit cast.